assets/src/components/imageUpload/index.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Image uploader component.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { Component, Fragment } from '@wordpress/element';
+import { Button } from '@wordpress/components';
+import { data } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import './style.scss';
+
+/**
+ * Image select/upload button and modal.
+ */
+class ImageUpload extends Component {
+
+	/**
+	 * Constructor.
+	 */
+	constructor( props ) {
+		super( ...arguments );
+		this.state = {
+			frame    : false,
+		}
+	}
+
+	/**
+	 * Open the WP media modal.
+	 */
+	openModal = () => {
+		if ( this.state.frame ) {
+			this.state.frame.open();
+			return;
+		}
+
+		this.state.frame = wp.media( {
+			title: __( 'Select or upload image' ),
+			button: {
+				text: __( 'Select' )
+			},
+			library: {
+				type: 'image'
+			},
+			multiple: false
+		} );
+
+		this.state.frame.on( 'select', this.handleImageSelect );
+		this.state.frame.open();
+	}
+
+	/**
+	 * Update the state when an image is selected from the media modal.
+	 */
+	handleImageSelect = () => {
+		const { onChange } = this.props;
+		const attachment = this.state.frame.state().get( 'selection' ).first().toJSON();
+		onChange( attachment );
+	}
+
+	/**
+	 * Clear the selected image.
+	 */
+	removeImage = () => {
+		const { onChange } = this.props;
+		onChange( null );
+	}
+
+	/**
+	 * Render.
+	 */
+	render = () => {
+		const { image } = this.props;
+
+		return (
+			<Fragment>
+				{ !! image && (
+				<div className="newspack-image-upload has-image">
+					<div className="image-preview">
+						<img src={ image.url } />
+					</div>
+					<Button className="remove-image" onClick={ this.removeImage }>
+						{ __( 'Remove image' ) }
```

Comment:
Regarding the styling of the `Remove image` button, I was wondering if the vertically centered placement came from Muriel, or if that was just how it happened to lay out when there's a placeholder image next to it?

Reply:
There was no design for that in Muriel or our designs, so I just "designed" that. Do you have a better-looking idea for that?

Reply:
No, not really—mostly just wanted to see if you had found it somewhere I wasn't looking, and if not to flag it for Sonja and Muriel folks to assess at a later date. Definitely no blocker here!